packages/ra-ui-materialui/src/list/filter/AddSavedQueryDialog.tsx

```diff
@@ -0,0 +1,101 @@
+import * as React from 'react';
+import { ReactElement, ChangeEvent, FormEvent, useState } from 'react';
+import { useListContext, useTranslate } from 'ra-core';
+import {
+    Button,
+    Dialog,
+    DialogActions,
+    DialogContent,
+    DialogTitle,
+    TextField,
+} from '@mui/material';
+
+import { useSavedQueries } from './useSavedQueries';
+
+export interface AddSavedQueryDialogProps {
+    open: boolean;
+    onClose: () => void;
+}
+
+export const AddSavedQueryDialog = ({
+    open,
+    onClose,
+}: AddSavedQueryDialogProps): ReactElement => {
+    const translate = useTranslate();
+    const {
+        resource,
+        filterValues,
+        displayedFilters,
+        sort,
+        perPage,
+    } = useListContext();
+
+    const [savedQueries, setSavedQueries] = useSavedQueries(resource);
+
+    // input state
+    const [queryName, setQueryName] = useState('');
+    const handleQueryNameChange = (
+        event: ChangeEvent<HTMLInputElement>
+    ): void => {
+        setQueryName(event.target.value);
+    };
+
+    const handleFormSubmit = (e: FormEvent<HTMLFormElement>): void => {
+        e.preventDefault();
+        addQuery();
+    };
+
+    const addQuery = (): void => {
+        setSavedQueries(
```

Comment:
Why wouldn't it be an array and what should I do if it's not?

Reply:
It can be something else if anotrher piece of code messes up with it. If it's not the right format, ignore the value and fall back to the default value. 

See https://github.com/marmelab/react-admin/blob/next/docs/Store.md#forward-compatibility for an explanation